karapace/coordinator/master_coordinator.py

```diff
@@ -42,6 +59,17 @@ def config(self) -> Config:
         return self._config
 
     async def start(self) -> None:
+        self._thread.start()
+
+    def _start_loop(self) -> None:
+        # we should avoid the reassignment otherwise we leak resources
+        assert self._loop is None, "Loop already started"
+        self._loop = asyncio.new_event_loop()
```

Comment:
The call below to set the event loop might have some sideeffects beyond this object, but I see the idea is to set this new loop within the spawned thread.
Do you think it is possible to use the `loop` already on the `app` object? 
If we can, we probably can use [loop.run_in_executor()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) to run the secluded code. 
We also could try to get the current running loop - `asyncio.get_running_loop()` and use the loop to run the threaded code, but it's also possible that at that time the application server hasn't started it's own loop.

Reply:
Not necessary, but I would prefer to have a single loop where it's lifecycle is already handled by the application server `aiohttp`, but I know you want to run it all in it's own thread, which leaves it properly isolated, so it's fine and you close the loop also 👍